packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,375 @@
+
+## Out of scope
+We don't have enough use-cases for the following features at Razorpay and hence scoped them out of our Table component. If you are aware of any of these features being used at Razorpay, please let us know by creating an issue - we will evaluate it and scope it for future releases.
+
+- Column Reordering
+- Column Resizing
+- Column Filtering
+- Search
+- Row Expansion
+- Nested Tables
+- Hiding Columns
+- Editable Rows
+
+## Anatomy
+<img src="./table-anatomy.png" width="760" alt="Table thumbnail" />
+
+## API
+
+### API Design Approaches
+While evaluating what kind of API we want for our table, we discovered 2 approaches:
+
+#### 1. Composable API
+Composable API is a pattern where we have a set of components that can be composed together to create a table. This approach allows us to stitch together a table very similar to how you would do it with native HTML. However, it is also more verbose and requires more effort to create a table.
+
+```jsx
+// Mock data & functions
+
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const onSortChange = ({ headerKey, sortType }) => {
+  console.log({headerKey, sortType});
+};
+
+
+```
+
+
+```jsx
+// Composable Table
+
+<Table 
+    data={data} // This contains the actual data to be rendered in the table which would be retrieved from some API
+    selectionType='single' 
+    onSelectionChange={onSelectionChange} 
+    sortFunctions={sortFunctions} 
+    onSortChange={onSortChange}
+    isLoading={false}
+    cellDensity='normal'
+    showZebraStripes={true}
+    pagination={{
+      limit: 10,
+      offset: 0,
+    }}
+>
+{(tableData) => (
+  <>
+    <TableToolbar title="Users" itemsSelectedTitle={`${selectedUsersCount} Users selected`}>
+        <ToolbarActions>
+            <Button>Export</Button>
+        </ToolbarActions>
+    </TableToolbar>
+
+    <TableHeader isSticky={true}>
+      <TableHeaderRow>
+         <TableHeaderCell headerKey='firstName'>
+          First Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='lastName'>
+          Last Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='balance'>
+          Balance
+          <Tooltip content='Current bank balance'>
+            <InfoIcon size='medium' />
+          </Tooltip>
+        </TableHeaderCell>
+       </TableHeaderRow>
+     </TableHeader>
+
+    <TableBody>
+      <TableRow onClick={console.log} isDisabled={false}>
+        <TableCell>
+          {tableData.firstName}
+        </TableCell>
+        <TableCell>
+          {tableData.lastName}
+        </TableCell>
+        <TableCell>
+          {tableData.balance}
+          <Badge>Low</Badge>
+        </TableCell>
```

Comment:
to enhance the reading the devs in real will put it as iteration right? so it won't look that verbose

Reply:
This section will always be verbose. If you see there is no "iteration" happening anywhere for the data. This defining what a single row looks like so balance will always have a badge in this case